substrate/frame/referenda/src/lib.rs

```diff
@@ -613,7 +627,11 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let now = frame_system::Pallet::<T>::block_number();
-			let mut status = Self::ensure_ongoing(index)?;
+
+			let mut status = match ReferendumInfoFor::<T, I>::get(index) {
+				Some(ReferendumInfo::Ongoing(status)) => Ok(status),
+				_ => Err(Error::<T, I>::NotOngoing),
```

Comment:
I think we should leave the `ensure_ongoing` as-is and do checks of being in the right state internally(are you not doing this/similar check already inside `ensure_ongoing`? Also I think we should consider candle period as part of Ongoing, it gives me the impression it is not?

Reply:
It is not. The main difference lies in the fact that you should not be able to commit (i.e. vote) in the confirm candle period. If we allow to show it as a logical part of the ongoing state (even though it is _"physically"_ as a trick to avoid a hard migration), that'd mean people could be allowed to modify the state of it (or at least, be in direct contradiction with the `NotOngoing` error on `access_poll`/`try_access_poll` methods returning that error.